pytorch_lightning/core/saving.py

```diff
@@ -202,7 +202,11 @@ def _load_model_state(cls, checkpoint: Dict[str, Any], strict: bool = True, **cl
         model.on_load_checkpoint(checkpoint)
 
         # load the state_dict on the model automatically
-        model.load_state_dict(checkpoint['state_dict'], strict=strict)
+        keys = model.load_state_dict(checkpoint['state_dict'], strict=strict)
+
+        if not strict:
+            rank_zero_warn(f"Missing Keys: {keys.missing_keys}")
+            rank_zero_warn(f"Unexpected Keys: {keys.unexpected_keys}")
```

Comment:
Can we replace `missing` and `unexpected` with their noob-friendly explanation? As described here

https://github.com/PyTorchLightning/pytorch-lightning/issues/8119#issuecomment-868847933

Reply:
@carmocca 
Do you mean like this ?
```
rank_zero_warn(f"Keys that are in model state dict but not in checkpoint: {keys.missing_keys}")
rank_zero_warn(f"Keys that are not in model state dict but in checkpoint: {keys.unexpected_keys}")
```

Reply:
Yes. "Found keys that are ..."

Also we should only print this if they are not empty

Reply:
I have added the changes along with the condition to print only is the list is non-empty.